domenetjenester/kompletthet/src/main/java/no/nav/foreldrepenger/kompletthet/impl/ManglendeInntektsmeldingTjeneste.java

```diff
@@ -132,23 +132,32 @@ public List<ManglendeVedlegg> utledManglendeInntektsmeldingerFraGrunnlag(Behandl
     public Optional<LocalDateTime> finnVentefristTilManglendeInntektsmelding(BehandlingReferanse ref, Skjæringstidspunkt skjæringstidspunkt) {
         var behandlingId = ref.behandlingId();
         var permisjonsstart = skjæringstidspunkt.getUtledetSkjæringstidspunkt();
-        var muligFrist = permisjonsstart.minusWeeks(TIDLIGST_VENTEFRIST_FØR_UTTAKSDATO_UKER);
+        // Blir brukt dersom søkt tidligere enn 4u før start ytelse
+        var muligFrist = permisjonsstart.minus(TIDLIGST_VENTEFRIST_IM_FØR_UTTAKSDATO);
+        // Brukes dersom søkt senere enn 4u før start ytelse
         var annenMuligFrist = søknadRepository.hentSøknadHvisEksisterer(behandlingId)
-            .map(s -> s.getMottattDato().plusWeeks(VENTEFRIST_ETTER_MOTATT_DATO_UKER));
-        var ønsketFrist = annenMuligFrist.filter(muligFrist::isBefore).orElse(muligFrist);
+            .map(s -> s.getMottattDato().plus(VENTEFRIST_IM_ETTER_SØKNAD_MOTTATT_DATO))
+            .orElse(muligFrist);
+        // Velg seneste frist av de to mulige
+        var ønsketFrist = annenMuligFrist.isAfter(muligFrist) ? annenMuligFrist : muligFrist;
         return finnVentefrist(ønsketFrist);
     }
 
     public Optional<LocalDateTime> finnVentefristForEtterlysning(BehandlingReferanse ref, Skjæringstidspunkt stp) {
         var behandlingId = ref.behandlingId();
         var permisjonsstart = stp.getUtledetSkjæringstidspunkt();
-        var muligFrist = LocalDate.now().isBefore(permisjonsstart.minusWeeks(TIDLIGST_VENTEFRIST_FØR_UTTAKSDATO_UKER)) ?
-            LocalDate.now() : permisjonsstart.minusWeeks(TIDLIGST_VENTEFRIST_FØR_UTTAKSDATO_UKER);
+        // Blir brukt dersom søkt tidligere enn 4u før start ytelse
+        var muligFrist = LocalDate.now().isBefore(permisjonsstart.minus(TIDLIGST_VENTEFRIST_IM_FØR_UTTAKSDATO)) ?
```

Comment:
for lesbarheten kan denne renames til muligFristHvisSøktForTidlig? eller kanskje bare fristSøktForTidlig